halva/src/main/java/io/soabase/halva/comprehension/MonadicFor.java

```diff
@@ -0,0 +1,43 @@
+package io.soabase.halva.comprehension;
+
+import io.soabase.halva.any.AnyVal;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.Function;
+
+/** Interface class for the non-type-safe version "for comprehension" for Monadic classes

```

Comment:
Is it really non-type-safe? I'm not sure. I think it's just as type safe as my implementation.